src/coprocessor/dag/rpn_expr/types.rs

```diff
@@ -0,0 +1,556 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use cop_datatype::{EvalType, FieldTypeAccessor};
+use tipb::expression::{Expr, ExprType, FieldType};
+
+use super::function::RpnFunction;
+use coprocessor::codec::batch::LazyBatchColumnVec;
+use coprocessor::codec::mysql::Tz;
+use coprocessor::dag::expr::EvalConfig;
+use coprocessor::data_type::{ScalarValue, ScalarValueRef, VectorValue, VectorValueRef};
+
+/// Global variables needed in evaluating RPN expression.
+#[derive(Debug)]
+pub struct RpnExpressionEvalContext {
```

Comment:
Because we need different context mechanisms (in future). For example, we may want to collect warnings and the row index that emits the warning (instead of just collecting the warnings). If we use `EvalContext` directly, then there will be no way to support both batch and non-batch, or we have to carefully make sure that our change doesn't break the non-batch version when modifying `EvalContext` to support the batch version. In either way, it looks to be more work to do. So I just created another infra, specifically for the batch version and does not affect the non-batch version at all.